compat/src/suspense.js

```diff
@@ -41,79 +42,14 @@ options.unmount = function(internal) {
 	if (oldUnmount) oldUnmount(internal);
 };
 
-/**
- *
- * @param {import('./internal').Internal} internal
- * @param {import('./internal').PreactElement} detachedParent
- * @param {import('./internal').PreactElement} parentDom
- */
-function detachedClone(internal, detachedParent, parentDom) {
-	if (internal) {
-		if (internal._component && internal._component.__hooks) {
-			internal._component.__hooks._list.forEach(effect => {
-				if (typeof effect._cleanup == 'function') effect._cleanup();
-			});
-
-			internal._component.__hooks = null;
-		}
```

Comment:
Hmm I'm not sure why we are clearing all hook state here. I don't think it makes sense for a `<Counter />`  component to have it's state cleared when a sibling suspends. And we only clear hook state and not class component state which seems inconsistent.

For context here is the related issue (#2795) and PR (#2796). Let me know if I'm missing a use case or something! :)

FWIW, React does not clear hook state or run effect cleanups when suspending (I copied our relevant tests into these code sandboxes:
* https://codesandbox.io/s/does-react-reset-hooks-when-suspending-mm1xn
* https://codesandbox.io/s/does-react-clean-effects-when-suspending-7pmri